modules/RTC/RTCUtils.js

```diff
@@ -114,9 +114,14 @@ function getConstraints(um, options) {
         constraints.video = { mandatory: {}, optional: [] };
 
         if (options.cameraDeviceId) {
-            // new style of settings device id (FF only)
-            constraints.video.deviceId = options.cameraDeviceId;
-            // old style
+            // Don't mix new and old style settings for Chromium as this leads
+            // to TypeError in new Chromium versions. @see
+            // https://bugs.chromium.org/p/chromium/issues/detail?id=614716
+            if (!RTCBrowserType.isChrome() && !RTCBrowserType.isOpera()) {
+                // New style of setting device id.
+                constraints.video.deviceId = options.cameraDeviceId;
+            }
+            // Old style.
             constraints.video.optional.push({
```

Comment:
To start on a humorous note, the Internet says: There's nothing more permanent than a temporary fix. (You've pointed out that there's a PR but, as you said, it's not ready yet.)

Do I understand you right that you don't want to use an if/else in the 3 cases of this PR?